### PR TITLE
add return to launch point agent

### PR DESCRIPTION
**Pull request by noodlemoodle**

### CODE REVIEW

File: src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ],
+
+  implements: [
+    'foam.core.ContextAgent'
+  ],
+
+  methods: [
+    async function execute() {
+      if ( this.currentMenu.id === this.ctrl.route ) {
+        window.history.back();
```
Comment (sarthak-marwaha) [inline src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js]: Will this work if the wizard is the first thing in browser history?

Comment (kgrgreer) [reply src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js]: When is this not true?

Comment (sarthak-marwaha) [reply src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js]: If you open a tab and go straight to the transferwizard, could be a link from an email or a direct link to onboarding

File: src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  implements: [
+    'foam.core.ContextAgent'
+  ],
+
+  methods: [
+    async function execute() {
+      if ( this.isIframe() ) return;
+      if ( this.currentMenu.id === this.ctrl.route ) {
+        if ( this.stack.pos >= 0 && window.history.length > 1 )
```
Comment (sarthak-marwaha) [inline src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js]: stack is no longer a reliable way to check the place in the app
```suggestion
        if ( window.history.length > 1 )
```

File: src/foam/u2/wizard/agents/StepWizardAgent.js [hunk 3]
```diff
@@ -64,11 +63,8 @@ foam.CLASS({
         if ( v == this.WizardStatus.IN_PROGRESS ) return;
         this.resolveAgent();
         let closePromise = this.wizardController.onClose();
-        if ( closePromise?.then ) {
-          closePromise.then(self.pushDefault)
-        } else {
-          self.pushDefault()
-        }
+        if ( closePromise?.then )
+          closePromise.then(() => {})
```
Comment (sarthak-marwaha) [inline src/foam/u2/wizard/agents/StepWizardAgent.js:67]: Can we not just remove this line?

Comment (noodlemoodle) [reply src/foam/u2/wizard/agents/StepWizardAgent.js:67]: we still use onclose in other places

File: src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js (hunk 2 above)
Comment (noodlemoodle) [reply src/foam/u2/wizard/agents/ReturnToLaunchPointAgent.js]: thanks, updated

**Pull request merged**